components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceOrganizationServiceImpl.java

```diff
@@ -0,0 +1,250 @@
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+
+    @GET
+    @Path("/{parentId}/children")
+    @Override
+    public Response getChildrenByParentId(@PathParam("parentId") String parentId) {
+        if (parentId == null) {
+            String errorMessage = "The parameter of the parent ID is empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        List<DeviceOrganizationMetadataHolder> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            result = dops.getChildrenByParentId(parentId);
+            if (result == null) {
+                String msg = "No children connected to device";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while retrieving children of device";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GET
+    @Path("/nodes")
+    @Override
+    public Response generateNodes() {
+        List<DeviceOrganizationVisNode> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        result = dops.generateNodes();
+        if (result.isEmpty()) {
+            String msg = "No Devices enrolled and visible";
+            log.error(msg);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        }
+        return Response.status(Response.Status.OK).entity(result).build();
+    }
+
+    @GET
+    @Path("/edges")
+    @Override
+    public Response generateEdges() {
+        List<DeviceOrganizationVisEdge> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        result = dops.generateEdges();
+        if (result.isEmpty()) {
+            String msg = "No Edge connections available and visible";
+            log.error(msg);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        }
+        return Response.status(Response.Status.OK).entity(result).build();
+    }
+
+    @GET
+    @Path("/hierarchy")
+    @Override
+    public Response generateHierarchy() {
+        return null;
```

Comment:
why are we returning `null` here? if this is not yet implemented better to use HTTP Status 501 - Not Implemented

Reply:
This code is for an unimplemented method for a propsed feature 768c0a7b2f49af30002236fefd8a02ed9acefe89. It has been removed.